jenkins.groovy

```diff
@@ -0,0 +1,132 @@
+/**
+ * Function for setup the test cluster.
+ */
+def setupEnvironment(String openshift) {
+    sh "rm -rf ~/.kube"
+    sh "sudo yum install -y zip"
+    clearImages()
+
+    def status = sh(
+            script: "oc cluster down",
+            returnStatus: true
+    )
+    if (status == 0) {
+        sh "for i in \$(mount | grep openshift | awk '{ print \$3}'); do sudo umount \"\$i\"; done && sudo rm -rf /var/lib/origin"
+        sh "sudo rm -rf /usr/bin/oc"
+    }
+
+    sh "for i in \$(mount | grep openshift | awk '{ print \$3}'); do sudo umount \"\$i\"; done && sudo rm -rf /var/lib/origin"
+    sh "sudo rm -rf /usr/bin/oc"
+
+    sh "mkdir -p /tmp/openshift"
+
+    status = sh(
+            script: "wget $openshift -O openshift.tar.gz",
+            returnStatus: true
+    )
+
+    timeout(time: 10, unit: 'MINUTES') {
+        if (status != 0) {
+            status = sh(
+                    script: "wget $openshift -O openshift.tar.gz",
+                    returnStatus: true
+            )
+        }
+    }
+    //////////////////////////////////////////////////
+    sh "tar xzf openshift.tar.gz -C /tmp/openshift --strip-components 1"
+    sh "sudo cp /tmp/openshift/oc /usr/bin/oc"
+    sh "sudo rm -rf /tmp/openshift/"
+    sh "sudo rm -rf openshift.tar.gz"
+
+    status = sh(
+            script: "oc cluster up --base-dir $WORKSPACE/origin/ --enable=*,service-catalog,web-console --insecure-skip-tls-verify=true",
+            returnStatus: true
+    )
+
+    if (status != 0) {
+        sleep(10)
+        sh "oc cluster down"
+        sh "oc cluster up --base-dir $WORKSPACE/origin/ --enable=*,service-catalog,web-console --insecure-skip-tls-verify=true"
```

Comment:
What will be here if "oc cluster up" fails again? I guess the script will execute the next steps... but it looks like a wrong decision. I suppose we need to execute "oc cluster up" again or we need to fail this Jenkins build with some message.

Reply:
The stage will fail -> build will fail as well and Jenkins will send email notification that build failed

Reply:
Can you please share a line where we have the second check for exit code? I see only first check at the line `if (status != 0)` but then if `sh "oc cluster up --base-dir $WORKSPACE/origin/ --enable=*,service-catalog,web-console --insecure-skip-tls-verify=true"` also fails we just go out from if block and execute next lines.

Reply:
Jenkins itself do this check. If some code exit with different ecode than 0 it will fail the stage and job if you don't have it in try/catch/error or check status like I do (as far as I know).